tests/OutputTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+namespace Dompdf\Tests;
+
+use CallbackFilterIterator;
+use Dompdf\Tests\TestCase;
+use Dompdf\Tests\OutputTest\Dataset;
+use FilesystemIterator;
+use Iterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SplFileInfo;
+
+final class OutputTest extends TestCase
+{
+    public const FAILED_OUTPUT_DIRECTORY = __DIR__ . "/../tmp/failed-output-tests";
+
+    /**
+     * @return Iterator<Dataset>
+     */
+    public static function datasets(): Iterator
+    {
+        $flags = FilesystemIterator::KEY_AS_FILENAME
+            | FilesystemIterator::CURRENT_AS_FILEINFO
+            | FilesystemIterator::SKIP_DOTS;
+        $filter = function (SplFileInfo $file) {
+            return $file->getExtension() === "html";
+        };
+        $dir = new RecursiveDirectoryIterator(Dataset::DATASET_DIRECTORY, $flags);
+        $files = new CallbackFilterIterator(new RecursiveIteratorIterator($dir), $filter);
+
+        foreach ($files as $file) {
+            $dataset = new Dataset($file);
+            yield $dataset->name() => $dataset;
+        }
+    }
+
+    public function outputTestProvider(): Iterator
```

Comment:
This could be merged with `datasets()`.

Reply:
Kept these separate because this is the data-provider version of the method. Merging the two would make the usage in `update-reference-output.php` a bit awkward.